backend/src/routes/userRoute.ts

```diff
@@ -0,0 +1,19 @@
+import { Router } from "express";
+import { getUserProfile,assignCoach, get_coaches, get_trainees, updateUserProfile, deleteUser, editUser, get_users, editTrainee, get_my_trainees } from "../controllers/auth.controller";
+import { verifyJWT } from "../middlewares/authenticate";
+
+
+const router = Router()
+
+router.get("/trainees",verifyJWT, get_trainees)
+router.get("/my-trainees",verifyJWT, get_my_trainees)
+router.get("/coaches",verifyJWT, get_coaches)
+router.get("/all",verifyJWT, get_users)
+router.patch("/assign-coach/:id",verifyJWT, assignCoach)
+router.put("/edit-coach-or-admin/:id", verifyJWT, editUser)
+router.put("/edit-trainee/:id", verifyJWT, editTrainee)
+router.get("/my-profile", verifyJWT, getUserProfile);
+router.put("/my-profile", verifyJWT, updateUserProfile); 
+router.delete('/:userId', deleteUser);
+
+export default router
```

Comment:
Can you try to be consistency in function naming like update_user_profile you can choose all to be kebab or camel case